### PR TITLE
ast: refactor getting and setting fields

### PR DESCRIPTION
**Pull request by kitbellew**
Helps with #2881.

### CODE REVIEW

File: scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala [hunk 1]
```diff
@@ -106,42 +108,46 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
...
         val paramss = rawparamss
 
         // step 4: turn all parameters into vars, create getters and setters
-        def internalize(name: TermName) = TermName("_" + name.toString)
         val fieldParamss = paramss
-        val fieldParams = fieldParamss.flatten.map(p => (p, p.name.decodedName.toString))
-        istats1 ++= fieldParams.map({ case (p, _) =>
-          var getterAnns = List(q"new $AstMetadataModule.astField")
-          if (p.mods.annotations.exists(_.toString.contains("auxiliary")))
-            getterAnns :+= q"new $AstMetadataModule.auxiliary"
```
Comment (tgodzik) [inline scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala:127]: So `auxiliary` was never used? Maybe remove it altogether?

**Review by tgodzik — APPROVED** [review]
LGTM, just one small comment, but otherwise it looks like simplification of the current code. Thanks for doing it!

File: scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala (hunk 1 above)
Comment (kitbellew) [reply scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala:127]: i assumed that it was looking for AstMetadata.auxilliary and not just any old annotation, and simply kept all annotations instead.

File: scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala [hunk 2]
```diff
@@ -106,42 +108,46 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
...
+        val fieldParams = fieldParamss.flatten
+        fieldParams.foreach { p =>
+          val getterAnns = q"new $AstMetadataModule.astField" :: p.mods.annotations
```
Comment (kitbellew) [inline scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala:139]: here, all annotations. but i don't know if any specific one is used.

**Pull request merged**